srv/components/controller/files/scripts/provision.sh

```diff
@@ -1010,6 +1023,55 @@ ftp_enable()
     } && echo "ftp_enable(): ftp is already enabled" >> $logfile
 }
 
+# Check if sftp service is enabled on controller
+is_sftp_enabled()
```

Comment:
Few issues with this implementation 
1. tmpfile is not removed at end in all cases
2. return value is not consistent. Sometimes it prints Enabled and in some cases it does not. 
3. Results are not logged in all the cases.

To simply, this function can return 0 or non zero depending on SFTP is enabled or not. 

```
# Check if sftp service is enabled on controller
# Return 0 if SFTP is enabled, else 1
{
    _ret=1
    _tmp_file="$tmpdir/is_sftp_enabled"
    rm -rf $_tmp_file

    # objects name in the xml
    _xml_obj_bt="security-communications-protocols"
    _xml_obj_plist=("sftp")

    # run command to get the details of advanced settings params
    echo "is_sftp_enabled():Checking if sftp service is enabled" >> $logfile
    cmd_run "show protocols"

    # parse xml to get required values of properties
    parse_xml $xml_doc $_xml_obj_bt "${_xml_obj_plist[@]}" > $_tmp_file
    [ -s $_tmp_file ] && grep -q "Enabled" $_tmp_file && _ret=0

    echo "is_sftp_enabled(): sftp result $ret" >> $logfile
    rm -rf $_tmp_file
    return $_ret
}
```

Reply:
Addressed this in the latest patch, please review.